graphql_compiler/compiler/compiler_frontend.py

```diff
@@ -456,8 +491,19 @@ def _compile_vertex_ast(schema, current_schema_type, ast,
             basic_blocks.append(fold_block)
             context['fold'] = inner_location
         elif recurse_directive:
+            # Get the valid fields of the outer scope, and the fields used in the inner scope.
+            outer_scope_type_map = schema.get_type_map().get(current_schema_type.name)
+            outer_scope_possible_properties = list(outer_scope_type_map.fields)
+            inner_scope_used_properties = [
```

Comment:
I'm not sure I understand why the outer scope and inner scope properties are important here -- aren't we just doing a type check? If the types work out, the properties must work out too, and if the types don't work out, we don't allow duck-typing based on the properties.

Reply:
good point, I think I swapped my supertypes and subtypes (i.e. this would imply that if you start on a subtype and then recurse on a supertype, then you could end up plucking a supertype property which the subtype doesn't have. Which makes no sense...